docs/glossary.rst

```diff
@@ -107,6 +107,15 @@ Glossary
         This is a term used to describe an operation where the user must
         ensure that the input is correct. Failure to do so can result in
         crashes, hangs, and other security issues.
+    
+    thread safety
+        All immutable objects in ``cryptography`` are safe to use in
+        multi-threaded environments. This means they can be shared across
+        threads without requiring additional synchronization. Mutable objects,
+        such as hash contexts, can also be shared, but concurrent modification
+        may lead to undefined behavior or exceptions. When working with
```

Comment:
Let's avoid the phrase undefined behavior, as it implies the possibility of security issues.

How about "exceptions or incorrect results."

Reply:
Updated